crates/rpc/rpc/src/otterscan.rs

```diff
@@ -205,7 +211,69 @@ where
     }
 
     /// Handler for `getContractCreator`
-    async fn get_contract_creator(&self, _address: Address) -> RpcResult<Option<ContractCreator>> {
-        Err(internal_rpc_err("unimplemented"))
+    async fn get_contract_creator(&self, address: Address) -> RpcResult<Option<ContractCreator>> {
+        if !self.has_code(address, None).await? {
+            return Ok(None);
+        }
+
+        // use binary search from block [1, latest block number] to find the first block where the
+        // contract was deployed
+        let mut low = 1;
+        let mut high = self.eth.block_number()?.try_into().unwrap();
```

Comment:
let's to saturating_to and not unwrap

Reply:
btw, @mattsse currently we need to trace all the transactions in the found block, with the `until` condition here only is the tx index. So can we extend this `until` condition and early break it when a condition is reached?

Reply:
oh I think you're looking for:

https://github.com/paradigmxyz/reth/blob/82770b0922380584e17ecd91cd06d9d0f81a16df/crates/rpc/rpc-eth-api/src/helpers/trace.rs#L119-L128

Reply:
thanks for it, dive into it

Reply:
It seems this is not what I want. this function will first seek the state to the pre of the txhash, then execute. if we apply this function for all txs in block, more things to do https://github.com/paradigmxyz/reth/blob/4b0cbe346b4a0280e53fccb8b209fb5c663b9d9e/crates/rpc/rpc-eth-api/src/helpers/trace.rs#L190-L197


what I want is something like below:

```
pre_state = pre_block.post_state
for tx in block.txes:
    result, post_state = trace_tx(tx, pre_state)
    if cond(result):
        break
    pre_state = post_state
```

never mind, this requirement is not very much, we can consider it where other needs.